pkg/v1/tkg/test/tkgctl/shared/ceip.go

```diff
@@ -118,3 +133,83 @@ func verifyTelemetryJobURL(context context.Context, url string, mcProxy *framewo
 
 	return errors.New("URL not found in the telemetry cron job")
 }
+
+func verifyTelemetryJobRunning(context context.Context, mcProxy *framework.ClusterProxy) error {
+	var (
+		err          error
+		selectors    = []crtclient.ListOption{crtclient.InNamespace(telemetryNamespace)}
+		client       = mcProxy.GetClient()
+		pollInterval = 30 * time.Second
+		pollTimeout  = 90 * time.Second
+	)
+
+	scheme := mcProxy.GetScheme()
+	batchv1.AddToScheme(scheme)
+
+	cronJob := &v1beta1.CronJob{}
+	if err = client.Get(context, types.NamespacedName{Name: telemetryName, Namespace: telemetryNamespace}, cronJob); err != nil {
+		return err
+	}
+	// updating the telemetry cron job schedule to "* * * * *" so that the cronjob can be scheduled to run within the next 59 seconds
+	cronJob.Spec.Schedule = "* * * * *"
+	if err = client.Update(context, cronJob); err != nil {
+		return err
+	}
+
+	// check to see if any telemetry job gets created within pollTimeout time interval
+	jobs := &batchv1.JobList{}
+	if err = wait.Poll(pollInterval, pollTimeout, func() (done bool, err error) {
+		if err = client.List(context, jobs, selectors...); err != nil {
+			if k8serr.IsNotFound(err) {
+				return false, nil
+			}
+			return false, err
+		}
+		if len(jobs.Items) == 0 {
+			return false, nil
+		}
+		return true, nil
+	}); err != nil {
+		return err
+	}
+	if len(jobs.Items) == 0 {
+		return errors.New("no telemetry job is running")
+	}
+
+	// check to see if any telemetry pod gets created within pollTimeout time interval
+	pods := &v1.PodList{}
+	if err = wait.Poll(pollInterval, pollTimeout, func() (done bool, err error) {
+		if err = client.List(context, pods, selectors...); err != nil {
+			if k8serr.IsNotFound(err) {
+				return false, nil
+			}
+			return false, err
+		}
+		if len(pods.Items) == 0 {
+			return false, nil
+		}
+		return true, nil
+	}); err != nil {
+		return err
+	}
+	if len(pods.Items) == 0 {
+		return errors.New("no telemetry pod is running")
+	}
+
+	// check to make sure that the telemetry pod does not have "Failed" status
+	if pods.Items[0].Status.Phase == "Failed" {
```

Comment:
I would check if the pod phase is running and this should use a wait with poll or preferably gomega's `Eventually` assertion.

Reply:
Sure, will change to only allow either of  'running' or 'succeeded' statuses